frontend/packages/shared/src/components/GiteaHeader/ThreeDotsMenu/LocalChangesModal/LocalChangesModal.test.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { render as rtlRender, screen, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { textMock } from '../../../../../../../testing/mocks/i18nMock';
+import { createQueryClientMock } from 'app-shared/mocks/queryClientMock';
+import { QueryClient } from '@tanstack/react-query';
+import { ServicesContextProps, ServicesContextProvider } from 'app-shared/contexts/ServicesContext';
+import { queriesMock } from 'app-shared/mocks/queriesMock';
+import { MemoryRouter } from 'react-router-dom';
+import { LocalChangesModal, LocalChangesModalProps } from './LocalChangesModal';
+
+describe('LocalChangesModal', () => {
+  const user = userEvent.setup();
+
+  it('calls onClose when the close button is clicked', async () => {
+    render();
+
+    const closeButton = screen.getByRole('button', { name: textMock('modal.close_icon') });
+    await act(() => user.click(closeButton));
+
+    expect(mockOnClose).toHaveBeenCalledTimes(1);
+  });
+
+  it('opens and closes the modal based on isOpen prop', () => {
+    render({ isOpen: false });
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+
+    render({ isOpen: true });
+    expect(screen.getByRole('dialog')).toBeInTheDocument();
+  });
```

Comment:
Foreslår å dele denne testen i to. En grei tommelfingerregel er å kalle `render` én gang per test.